### PR TITLE
decode_b: Cleanup a chunk of code

### PR DESCRIPTION
**Pull request by randomPoison**
* Add helper methods to `mv` an `Av1Block` for accessing unnamed fields.

### CODE REVIEW

File: src/decode.rs [hunk 1]
```diff
@@ -4587,172 +4587,42 @@ unsafe fn decode_b(
...
-                        .warpmv
-                        .matrix[5 as libc::c_int
-                        as usize] = b
-                        .c2rust_unnamed
-                        .c2rust_unnamed_0
-                        .c2rust_unnamed
-                        .c2rust_unnamed_0
-                        .matrix[3 as libc::c_int as usize] as libc::c_int
-                        + 0x10000 as libc::c_int;
+                    t.warpmv.matrix[2] = b.matrix()[0] as int32_t + 0x10000;
```
Comment (kkysen) [inline src/decode.rs:4595]: This one used to be `as c_int`, not `as int32_t`.

File: src/decode.rs [hunk 2]
```diff
@@ -4587,172 +4587,42 @@ unsafe fn decode_b(
...
+                    t.warpmv.matrix[3] = b.matrix()[1] as int32_t;
+                    t.warpmv.matrix[4] = b.matrix()[2] as int32_t;
+                    t.warpmv.matrix[5] = b.matrix()[3] as int32_t + 0x10000;
```
Comment (kkysen) [inline src/decode.rs:4598]: This one used to be `as c_int`, not `as int32_t`, too.

File: src/decode.rs [hunk 3]
```diff
@@ -4587,172 +4587,42 @@ unsafe fn decode_b(
...
-                            b
-                                .c2rust_unnamed
-                                .c2rust_unnamed_0
-                                .c2rust_unnamed
-                                .c2rust_unnamed_0
-                                .mv2d
-                                .c2rust_unnamed
-                                .x as libc::c_int,
+                        println!(
+                            "[ {:+x} {:+x} {:+x}\n  {:+x} {:+x} {:+x} ]\n\
```
Comment (kkysen) [inline src/decode.rs:4612]: The `signabs` macro expanded above is slightly different from this.  It prints `-` if negative or else ` `, not `+`, which I believe is what the `+` sign formatting flag does in Rust.  I think we should just deduplicate the `signabs` macro.

Comment (kkysen) [reply src/decode.rs:4612]: What does the `x` in `{:+x}` mean?

File: src/decode.rs (hunk 1 above)
Comment (randomPoison) [reply src/decode.rs:4595]: `t.warpmv.matrix` is an array of `int32_t`, so I'd prefer to cast directly to that type here.

File: src/decode.rs (hunk 3 above)
Comment (randomPoison) [reply src/decode.rs]: The `x` in `{:+x}` means that the value is formatted as a hexadecimal number.

You're right, I completely mistranslated what this print output was doing (I entirely forgot the `abs` invocation!). I've changed it to a new version that translates `signabs` into a Rust helper struct, it should behave closer to the original now.

File: src/levels.rs [hunk 4]
```diff
@@ -196,6 +196,16 @@ pub union mv {
...
 
+impl mv {
+    pub fn x(&self) -> i16 {
+        unsafe { self.c2rust_unnamed.x }
+    }
+
+    pub fn y(&self) -> i16 {
+        unsafe { self.c2rust_unnamed.y }
+    }
+}
```
[the remark this answers is not in the document]
Comment (randomPoison) [reply src/levels.rs:207]: Sure, seems like useful cleanup to do. How do you plant to make it fully safe though?

[the remark this answers is not in the document]
Comment (kkysen) [reply src/levels.rs:207]: Okay, I'll wait until after this merges.  I plan to make `x` and `y` getters (and setters will be needed, too, probably) for the `n: u32` field.  The union here seems to be only used for convenience.

File: src/decode.rs (hunk 2 above)
Comment (kkysen) [reply src/decode.rs:4598]: Resolved here: https://github.com/memorysafety/rav1d/pull/132#discussion_r1169106458.

File: src/decode.rs (hunk 3 above)
Comment (kkysen) [reply src/decode.rs]: a0c31084bb60ab3564603136ef06bc1f18eb0a71 looks good.

File: src/decode.rs [hunk 5]
```diff
@@ -4587,172 +4587,54 @@ unsafe fn decode_b(
...
+
+                        /// Helper struct for printing a number as a signed hexidecimal value.
+                        struct SignAbs(i32);
+
+                        impl fmt::Display for SignAbs {
+                            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                                let sign = if self.0 < 0 { "-" } else { " " };
+                                write!(f, "{}{:x}", sign, self.0.abs())
+                            }
+                        }
```
Comment (kkysen) [inline src/decode.rs:4621]: `signabs` was used twice in `decode_b`, though re-defined.  I think we can move this to the top of `decode_b`.

**Review by kkysen — APPROVED** [review]

**Pull request merged**